test/unit-test/gtclang/Parsing/TestParsing.cpp

```diff
@@ -0,0 +1,175 @@
+    return nf;
+  }
+
+  NestableFunctions addVerticalRegion(const Twine& min, const Twine& max) {
+    NestableFunctions nf(Twine::createNull(), "vertical_region", ss(), IndentLevel + 1);
+    nf.addArg(min).addArg(max);
+    nf.startBody();
+    return nf;
+  }
+
+  NestableFunctions addIfStatement(const Twine& ifCondition) {
+    NestableFunctions ifStatement(Twine::createNull(), "if", ss(), IndentLevel + 1);
+    ifStatement.addArg(ifCondition);
+    ifStatement.startBody();
+    return ifStatement;
+  }
+};
+
+struct VerticalRegion : public NestableFunctions {
+  VerticalRegion(const Twine& name, std::stringstream& s, int il = 0)
+      : NestableFunctions(Twine::createNull(), name, s, il) {}
+};
+
+struct DawnObject : public codegen::Structure {
+  using Structure::Structure;
+  DawnObject(const Twine& type, const Twine& name, std::stringstream& s)
+      : Structure(type.str().c_str(), name, s) {}
+
+  Statement addStorage(const Twine& memberName) {
+    Statement member(ss(), IndentLevel + 1);
+    member << "storage"
+           << " " << memberName;
+    return member;
+  }
+
+  NestableFunctions addDoMethod(const Twine& type) {
+    NestableFunctions nf(type, "Do", ss(), IndentLevel + 1);
+    return nf;
+  }
+
+  Statement addOffset(const Twine& offsetName) {
+    Statement member(ss(), IndentLevel + 1);
+    member << "offset"
+           << " " << offsetName;
+    return member;
+  }
+};
+
+struct StencilFunction : public DawnObject {
+  StencilFunction(const Twine& name, std::stringstream& s)
+      : DawnObject("stencil_function", name, s) {}
+
+  virtual NestableFunctions addDoMethod(const Twine& type) {
+    NestableFunctions nf(type, "Do", ss(), IndentLevel + 1);
+    nf.startBody();
+    return nf;
+  }
+};
+
+struct Stencil : public DawnObject {
+  Stencil(const Twine& name, std::stringstream& s) : DawnObject("stencil", name, s) {}
+};
+
+struct Globals : public DawnObject {
+  Globals(std::stringstream& s) : DawnObject("globals", Twine::createNull(), s) {}
+  virtual NestableFunctions addDoMethod(const Twine& type) = delete;
+  virtual Statement addOffset(const Twine& offsetName) = delete;
+};
+
+class FileWriter {
+public:
+  FileWriter(std::string filename) {
+    filename_ = env_.getFileManager().dataPath() + filename;
+    writeSkeleton();
+  }
+
+  void writeSkeleton() {
+    fileHeader = HeaderWriter::longheader();
+    fileHeader += HeaderWriter::includes();
+  }
+
+  std::stringstream& ss() { return ss_; }
+
+  void writeToFile() {
+    std::ofstream ofs(filename_, std::ofstream::trunc);
+    ofs << fileHeader;
+    ofs << ss_.str();
+    ofs.close();
+  }
+  const std::string getFileName() const { return filename_; }
+  UnittestEnvironment& getEnv() const { return env_; }
+
+private:
+  std::stringstream ss_;
+  std::string filename_;
+  std::string fileHeader;
+  UnittestEnvironment& env_ = UnittestEnvironment::getSingleton();
+};
+
+TEST(ParsingTest, Assignment_Stencil) {
```

Comment:
This is not really good unit testing style. I would go for a smaller **unit** tests which test a portion of the functionality each instead of one big test. I know it is painful to write tests but it will pay of in the long term ;)